### PR TITLE
make safe_embedding_lookup_sparse method public and clean duplicate codes

### PR DESCRIPTION
**Pull request by facaiy**
In #17417, I found that same `safe_embedding_lookup_sparse` code exists in both embedding_ops.py and feature_column.py.

To resolve circular dependency, I create an `embedding_ops_py` target. The method looks not so good, any suggestion would be appreciated.

### CODE REVIEW

**Review by ispirmustafa — CHANGES_REQUESTED** [review]
Thank you Yan,
Let's make the core version public, instead of making contrib version public. contrib.layers fcs will be deprecated soon.
Could you please make following one public?
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/feature_column/feature_column.py#L3069

**Comment by facaiy** [comment]
@ispirmustafa Thanks for your clarification. So let's move the method to `tensorflow/python/ops/embedding_ops.py` and make it public, right?

**Comment by facaiy** [comment]
@ispirmustafa Hi,  I have made the core version public, could you take a look again?

File: tensorflow/contrib/layers/python/layers/embedding_ops.py [hunk 1]
```diff
@@ -100,112 +98,15 @@ def safe_embedding_lookup_sparse(embedding_weights,
...
-
-
-def _prune_invalid_weights(sparse_ids, sparse_weights):
-  """Prune invalid weights (< 0) from the input ids and weights."""
-  if sparse_weights is not None:
-    is_weights_valid = math_ops.greater(sparse_weights.values, 0)
-    sparse_ids = sparse_ops.sparse_retain(sparse_ids, is_weights_valid)
-    sparse_weights = sparse_ops.sparse_retain(sparse_weights, is_weights_valid)
-  return sparse_ids, sparse_weights
+  return embedding_ops.safe_embedding_lookup_sparse(
```
Comment (ispirmustafa) [inline tensorflow/contrib/layers/python/layers/embedding_ops.py]: let's do not change the contrib one.

**Review by ispirmustafa — COMMENTED** [review]
thank you Yan.
a small comment, lgtm otherwise.

File: tensorflow/contrib/layers/python/layers/embedding_ops.py (hunk 1 above)
Comment (facaiy) [reply tensorflow/contrib/layers/python/layers/embedding_ops.py]: Fine, I have reverted the change in the contrib 😃

**Comment by facaiy** [comment]
@martinwicke Hi, could you make an API review? Thank you.

**Review by ispirmustafa — DISMISSED** [review]

**Comment by martinwicke** [comment]
This will fail the tests -- can you run the API golden update?

**Comment by facaiy** [comment]
Sure, API golden has been updated.

**Comment by ekelsen** [comment]
@ispirmustafa Could you finish the review please?

**Review by ispirmustafa — DISMISSED** [review]

**Comment by martinwicke** [comment]
(good for API review)

**Comment by facaiy** [comment]
Does anyone can help restart all tests? Thanks.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @ispirmustafa: It has been 18 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @ispirmustafa: It has been 32 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @ispirmustafa: It has been 47 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Review by ispirmustafa — APPROVED** [review]
Thank you!

**Comment by facaiy** [comment]
@martinwicke @ispirmustafa Thank you. All tests pass.

**Pull request merged**